File: backend-ai/src/lib/response.py
"""
Standard API response helpers for the Meridian AI backend.
"""
from datetime import datetime, timezone
from typing import Any


def success_response(data: Any, message: str = "Success") -> dict:
    """Creates a successful API response."""
    return {
        "success": True,
        "data": data,
        "message": message,
        "error": None,
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }


def error_response(message: str, error: str | None = None) -> dict:
    """Creates an error API response."""
    return {
        "success": False,
        "data": None,
        "message": message,
        "error": error,
        "timestamp": datetime.now(timezone.utc).isoformat(),
    }

File: backend/src/lib/response.ts
/**
 * Standard API response format for all Meridian endpoints.
 */
export interface ApiResponse<T> {
  success: boolean
  data: T | null
  message: string
  error: string | null
  timestamp: string
}

/**
 * Creates a successful API response.
 */
export function successResponse<T>(
  data: T,
  message = "Success"
): ApiResponse<T> {
  return {
    success: true,
    data,
    message,
    error: null,
    timestamp: new Date().toISOString(),
  }
}

/**
 * Creates an error API response.
 */
export function errorResponse(
  message: string,
  error: string | null = null
): ApiResponse<null> {
  return {
    success: false,
    data: null,
    message,
    error,
    timestamp: new Date().toISOString(),
  }
}

export const response = {
  success: successResponse,
  error: errorResponse,
}
